docs/user/Overview.md

```diff
@@ -0,0 +1,82 @@
+- `Machines`
+- `MachineSets`
+- `MachineHealthChecks`
+
+This operator is responsible for the creation and maintenance of:
+- `machine-api-operator` `ClusterOperator` - `MAO` status reporting
+- `machine-api-controllers` `Deployment` - controllers for all supported `CRDs`
+- `machine-api` `ValidatingWebhookConfiguration` and `MutatingWebhookConfiguration` - validation and defaulting for Machine resources
+termination handler `DaemonSet` - monitoring for spot instances state and remediating `Machines`, which are deployed on those in case the instance goes away.
+
+### Implementing
+
+- `Machine` controller - manages Machine resources. It uses actuator interface, which follows a `Machine` lifecycle [pattern](https://github.com/openshift/enhancements/blob/master/enhancements/machine-api/machine-instance-lifecycle.md) This interface provides `Create`, `Update`, and `Delete` methods to manage your provider specific cloud instances, connected storage, and networking settings to make the instance prepared for bootstrapping. Each provider is therefore responsible for implementing these methods.
+- `MachineSet` controller - manages `MachineSet` resources and ensures the presence of the expected number of replicas and a given provider config for a set of machines.
+- `MachineHealthCheck` controller - manages `MachineHealthCheck` resources. Ensure machines being targeted by `MachineHealthCheck` objects are satisfying healthiness criteria or are remediated otherwise.
+- `NodeLink` controller - ensure machines have a nodeRef based on `providerID` matching. Annotate nodes with a label containing the machine name.
+
+### Integrating 
+
+Providers which currently works with `MAO`, are:
+- [AWS](https://github.com/openshift/cluster-api-provider-aws)
+- [GCP](https://github.com/openshift/cluster-api-provider-gcp/)
+- [OpenStack](https://github.com/openshift/cluster-api-provider-openstack/)
+- [vSphere](https://github.com/openshift/machine-api-operator/tree/master/pkg/controller/vsphere)
+- [Azure](https://github.com/openshift/cluster-api-provider-azure)
+- [BareMetal](https://github.com/openshift/cluster-api-provider-baremetal/)
+- [OVirt](https://github.com/openshift/cluster-api-provider-ovirt)
+
+## Works closely, but not directly responsible for
+
+- [cluster-api-actuator-pkg](https://github.com/openshift/cluster-api-actuator-pkg/) - hosting e2e tests for `MAO` and supported cloud providers.
+machine-config-operator - creating machineConfigs with configuration to inject into provisioned Machine instances. Is responsible for initiating Node bootstrap procedure for newly created Machine.
+- [cluster-machine-approver](https://github.com/openshift/cluster-machine-approver) - approving Node CSR for newly provisioned Machine.
+- [cluster-autoscaler](https://github.com/openshift/kubernetes-autoscaler/tree/master/cluster-autoscaler) - automatic scaling of MachineSet resources, manages ClusterAutoscaler and MachineAutoscaler resources.
```

Comment:
> manages ClusterAutoscaler and MachineAutoscaler resources.

This is cluster-autoscaler-operator not cluster-autoscaler, should include that repo linked here as well